**SM339 &#x25aa; Applied Statistics &#x25aa; Spring 2023 &#x25aa; Uhan**

# Lesson 15. Coefficient of Determination for Multiple Linear Regression

## The coefficient of (multiple) determination $R^2$

- Question: __How much of the variation in $Y$ is explained by the model?__

- Formula:

$$ R^2 = \frac{\text{variability explained by model}}{\text{total variability in $Y$}} = \frac{\mathit{SSModel}}{\mathit{SSTotal}} = 1 - \frac{\mathit{SSE}}{\mathit{SSTotal}} $$

- Interpretation:

    - Suppose we fit a multiple linear regression model and the $R^2$ value is 0.75
    
    - Then we would say
        > The linear regression model with explanatory variables __[list of explanatory variables]__ explains 75% of the variability in the response variable __[response variable]__.

- Things to note:

    - We call $R^2$ the coefficient of _multiple_ determination because we now have multiple predictors
    
    - For simple linear regression, we use $r^2$ (with lowercase $r$) to denote the coefficient of determination because it was the sample correlation $r$ between $X$ and $Y$ squared
    
    - That interpretation does not translate directly to the case with multiple predictors, since each predictor has its own correlation with the response variable
    
    - However, in the multiple linear regression setting, if we calculate the correlation between $y$ (observed) and $\hat{y}$ (fitted) and then square that correlation, we would obtain $R^2$

### Example 1

Continuing with the `RailsTrails` data and model from the previous lesson...

In [1]:
library(Stat2Data)
data(RailsTrails)

#### a.
How much of the variability in $\mathit{Price2014}$ is explained by the _simple_ linear regression model with _only_ $\mathit{SquareFeet}$ as the predictor?

In [2]:
# Solution
fit <- lm(Price2014 ~ SquareFeet, data = RailsTrails)
summary(fit)


Call:
lm(formula = Price2014 ~ SquareFeet, data = RailsTrails)

Residuals:
    Min      1Q  Median      3Q     Max 
-172.13  -38.20   -2.05   28.33  330.20 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)    43.77      19.53   2.241   0.0272 *  
SquareFeet    159.17      11.75  13.544   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 66.61 on 102 degrees of freedom
Multiple R-squared:  0.6427,	Adjusted R-squared:  0.6392 
F-statistic: 183.5 on 1 and 102 DF,  p-value: < 2.2e-16


*Write your notes here. Double-click to edit.*

*Solution.* $R^2 = 0.6427$.

#### b.
How much of the variability in $\mathit{Price2014}$ is explained by the _simple_ linear regression model with _only_ $\mathit{Distance}$ as the predictor?

In [3]:
# Solution
fit <- lm(Price2014 ~ Distance, data = RailsTrails)
summary(fit)


Call:
lm(formula = Price2014 ~ Distance, data = RailsTrails)

Residuals:
    Min      1Q  Median      3Q     Max 
-202.74  -56.27  -15.91   31.53  551.45 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   352.16      15.18  23.192  < 2e-16 ***
Distance      -53.01      10.44  -5.079 1.72e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 99.55 on 102 degrees of freedom
Multiple R-squared:  0.2018,	Adjusted R-squared:  0.194 
F-statistic: 25.79 on 1 and 102 DF,  p-value: 1.72e-06


*Write your notes here. Double-click to edit.*

*Solution.* $R^2 = 0.2018$.

#### c.
How much of the variability in  $\mathit{Price2014}$ is explained by the _multiple_ linear regression model with both $\mathit{SquareFeet}$ and $\mathit{Distance}$ as predictors?

In [4]:
# Solution
fit <- lm(Price2014 ~ SquareFeet + Distance, data = RailsTrails)
summary(fit)


Call:
lm(formula = Price2014 ~ SquareFeet + Distance, data = RailsTrails)

Residuals:
    Min      1Q  Median      3Q     Max 
-152.15  -30.27   -4.14   25.75  337.93 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   78.985     25.607   3.085  0.00263 ** 
SquareFeet   147.920     12.765  11.588  < 2e-16 ***
Distance     -15.788      7.586  -2.081  0.03994 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 65.55 on 101 degrees of freedom
Multiple R-squared:  0.6574,	Adjusted R-squared:  0.6506 
F-statistic: 96.89 on 2 and 101 DF,  p-value: < 2.2e-16


*Write your notes here. Double-click to edit.*

*Solution.* $R^2 = 0.6574$.

## The adjusted coefficient of determination (adjusted $R^2$) 

- Adding an additional predictor to a multiple linear regression model can _never_ decrease the percentage of variability explained by that model (assuming we are using the same data)

- __But: is the increase in explained variability due to important new information in the new predictor, or is it just due to chance?__

- One way to answer this question is to do a $t$-test for the new predictor's coefficient, like we did in the previous lesson

- Another less formal option is to look at the __adjusted $R^2$__, denoted by $R_{adj}^2$

- The __adjusted coefficient of determination $R_{adj}^2$__ accounts for
    
    1. the amount of variability explained by the model, __and__
    
    2. the number of predictors
    
- It includes a _penalty_ for additional predictors, so an extra predictor has to help _enough_ in order to raise $R_{adj}^2$

- $R_{adj}^2$ can go _up or down_ when a new predictor is included

- Therefore, $R_{adj}^2$ is a good way to capture models with different numbers of predictors
    - It's most useful to quickly compare lots of models

- Formula:
    $$R_{adj}^2 = 1 - \frac{\hat{\sigma}_{\varepsilon}^2}{s_{Y}^2} = 1 - \frac{ \mathit{SSE} \,/\, (n - (k + 1)) }{ \mathit{SSTotal} \,/\, (n - 1) } $$
    
    - $s_{Y}^2$ is the sample variance of $Y$

### Example 2

Continuing with Example 1...

#### a.
For our two predictor model of house prices, what is $R^2_{adj}$? Look for "Adjusted R-squared" in the Example 1 output.

*Write your answer here. Double-click to edit.*

*Solution.* $R^2_{adj} = 0.6506$.

#### b.
What is $R_{adj}^2$ for the simple linear regression model using only $\mathit{SquareFeet}$ as a predictor of $\mathit{Price2014}$?

*Write your answer here. Double-click to edit.*

*Solution.* $R_{adj}^2 = 0.6392$.

#### c.
Based on your answers above, which model seems to be better?

*Write your answer here. Double-click to edit.*

*Solution.* The model with both predictors seems to better, since it has a larger $R_{adj}^2$. 